### PR TITLE
[PLAT-2169] Box query tool polish

### PR DESCRIPTION
**Pull request by jdm717**
## Summary

Adds some UX/interaction polish to the `<vertex-viewer-box-query-tool>` element, specifically:
- Corrects behavior when switching between `perspective` and `orthographic` camera types
- Adds cursors to indicate when an operation is in-flight
  - Additionally, this has now been limited to a single in-flight operation
- Fixes an issue where moving the mouse outside the viewer would cause the box to behave a bit erratically 
- Fixes an issue where the ability to draw additional boxes while an operation was in-flight would visually appear to work
- Adds support for a `mode` on the query tool, which can be set to `exclusive`, `inclusive`, or `undefined` to limit to either mode (with `undefined` being default/supporting both)

## Test Plan

- Test the above mentioned improvements/changes

## Release Notes

N/A

## Possible Regressions

Cursor management with the query tool, combined `exclusive`/`inclusive` query behavior

## Dependencies

N/A

### CODE REVIEW

File: packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.spec.tsx [hunk 1]
```diff
@@ -216,6 +217,91 @@ describe('vertex-viewer-box-query-tool', () => {
...
+      });
+
+    await drawInclusiveBox(page, viewer);
+    window.dispatchEvent(new MouseEvent('pointerup'));
+
+    await drawExclusiveBox(page, viewer);
+    window.dispatchEvent(new MouseEvent('pointerup'));
+
+    await page.waitForChanges();
+    await Async.delay(10);
```
Comment (MadisonEhlers-Vertex) [inline packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.spec.tsx:251]: Does the test fail without a manual delay?

File: packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx [hunk 2]
```diff
@@ -11,6 +11,8 @@ import {
 
 export type VolumeIntersectionQueryType = 'select' | 'deselect';
 
+export type VolumeIntersectionQueryMode = 'exclusive' | 'inclusive';
```
Comment (MadisonEhlers-Vertex) [inline packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx:14]: Should we consider adding a third option, `both`? Rather than having `undefined` function as both?

File: packages/viewer/src/lib/volume-intersection/controller.ts [hunk 3]
```diff
@@ -67,11 +73,14 @@ export class VolumeIntersectionQueryController {
...
+        this.operationInFlight = false;
+        this.executeComplete.emit();
       }
-    } else if (this.inFlightOperations >= this.maxInFlightOperations) {
+    } else if (this.operationInFlight) {
       throw new Error(
-        `Unable to perform volume intersection query due to the limit of ${this.maxInFlightOperations}.`
+        `Unable to perform volume intersection query as there is already one in-flight.`
       );
     }
```
Comment (MadisonEhlers-Vertex) [inline packages/viewer/src/lib/volume-intersection/controller.ts:85]: not related to this PR, but should we add an `else` here to handle if the screenBounds _are_ null? Log or throw an error or something?

File: packages/viewer/src/lib/volume-intersection/interactions.ts [hunk 4]
```diff
@@ -1,51 +1,104 @@
...
+    this.enabled = false;
+  }
+
+  public enable(): void {
+    this.enabled = true;
   }
 
   private handleDragBegin(event: PointerEvent): void {
-    if (event.buttons === 1 && !this.isInteracting) {
+    if (this.enabled && event.buttons === 1 && !this.isInteracting) {
```
Comment (MadisonEhlers-Vertex) [inline packages/viewer/src/lib/volume-intersection/interactions.ts:57]: 💡 Just an idea here, I often forget what all of the button numbers map to. Do you think it'd be a good idea to add some utils around the PointerEvent buttons to make it more readable? i.e. 

```ts
private function isPrimaryClick(event: PointerEvent): boolean {
  return event.buttons === 1;
}
```
Just an idea to make some of the code more readable. Not that important.

File: packages/viewer/src/lib/volume-intersection/interactions.ts [hunk 5]
```diff
@@ -1,51 +1,104 @@
...
+
+      this.crosshairCursorDisposable?.dispose();
+      this.addWaitCursor();
+      try {
+        await this.controller.execute();
+      } finally {
+        this.waitCursorDisposable?.dispose();
+        this.addCrosshairCursor();
+      }
+    }
```
Comment (MadisonEhlers-Vertex) [inline packages/viewer/src/lib/volume-intersection/interactions.ts:92]: What should happen if the drag ends when it's not enabled?

File: packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.spec.tsx (hunk 1 above)
Comment (jdm717) [reply packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.spec.tsx:251]: Yeah, unfortunately. At least the first delay is necessary to "complete" the first alteration and allow for the third attempt to complete successfully. Looks like I'm able to remove the second await though, so I'll do that

File: packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx (hunk 2 above)
Comment (jdm717) [reply packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx:14]: Yeah, I thought about having a `both`, but it felt like that didn't communicate enough about the types that would be present. Though the same argument could probably be made for `undefined`. I also considered `exclusive-and-inclusive`, but that felt a bit wordy. Open to ideas here 🙂

File: packages/viewer/src/lib/volume-intersection/controller.ts (hunk 3 above)
Comment (jdm717) [reply packages/viewer/src/lib/volume-intersection/controller.ts:85]: Sure, I can take a look at adding that. Probably at least worth an error

File: packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx (hunk 2 above)
Comment (mikemccarty-vertex) [reply packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx:14]: What would `both` do?

As defined right now:
`inclusive == In | Intersects`
`exclusive == In & ~(Intersects | Out)`

Comment (jdm717) [reply packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.tsx:14]: @mikemccarty-vertex the `both` would just refer to what queries the tool itself is capable of. I.e. with `both` as the query mode, dragging left-to-right would perform an `exclusive` query and right-to-left would perform an `inclusive` query. With this change, the `inclusive` and `exclusive` modes mean that no matter which way you drag, you'll get that type of query

**Review by MadisonEhlers-Vertex — APPROVED** [review]
